packages/core/integration-tests/test/javascript.js

```diff
@@ -2678,6 +2681,59 @@ describe('javascript', function () {
     });
   });
 
+  it('should replace __dirname and __filename with path.join calls relative to projectRoot', async function () {
+    let b = await bundle(
+      path.join(__dirname, '/integration/env-node-replacements/index.js'),
+    );
+
+    let dist = await outputFS.readFile(b.getBundles()[0].filePath, 'utf8');
+    assert(
+      dist.includes(
+        xplatPath(
+          'require("path").join(__dirname, "../test/integration/env-node-replacements")',
+        ),
+      ),
+    );
+    assert(
+      dist.includes(
+        xplatPath(
+          'require("path").join(__dirname, "../test/integration/env-node-replacements", "index.js")',
+        ),
+      ),
+    );
+    let f = await run(b);
+    let output = f();
+    assert.equal(output.data, 'hello');
+    assert(
+      output.firstDirnameTest.includes(
+        xplatPath(
+          'integration-tests/test/integration/env-node-replacements/data',
+        ),
+      ),
+    );
```

Comment:
I think something like
```js
assert.equals(output.firstDirnameTest, path.join(__dirname, '/integration/env-node-replacements/data'));
```
would be better here. That also covers using the appropriate path separators for the current platform

Reply:
@mischnic done